### PR TITLE
Start daemon if certificates have been expired

### LINKED ISSUE #24132: Daemon can't be started after swarm certificates expire
Swarm certificates automatically renew and have 90 day expiry period by default. Still, if you don't start the daemon during that time the certificates will expire and starting daemon will fail with `time="2016-06-29T17:18:06.165656736Z" level=fatal msg="Error creating cluster component: error while loading TLS Certificate in /var/lib/docker/swarm/certificates/swarm-node.crt: x509: certificate has expired or is not yet valid"` 

I think refusing to start and not ignoring this error is correct. We could provide `--reset-swarm` option to leave swarm so the user doesn't need to remove the state dir manually. Problem is that user must remember to remove this option as otherwise, it would clear the state on every next restart as well.

Maybe a good enough solution would be to add instructions for removing the state directory in the error message.

@nathanleclaire

### PR DESCRIPTION
**Pull request by tonistiigi**
fixes #24132

cc @aaronlehmann @diogomonica 

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

File: daemon/cluster/cluster.go [hunk 1]
```diff
@@ -927,13 +939,21 @@ func (c *Cluster) isActiveManager() bool {
 	return c.node != nil && c.conn != nil
 }
 
+// isSwarmExists should not be called without a read lock
+func (c *Cluster) isSwarmExists() bool {
```
Comment (aaronlehmann) [inline daemon/cluster/cluster.go]: nit: The name is a bit weird, I would use something like `swarmExists`.

**Comment by tonistiigi** [comment]
@aaronlehmann PTAL

**Comment by aaronlehmann** [comment]
LGTM

File: daemon/cluster/cluster.go [hunk 2]
```diff
@@ -186,7 +190,10 @@ func New(config Config) (*Cluster, error) {
 		if errors.Cause(c.err) == ErrSwarmLocked {
 			return c, nil
 		}
-
+		if errors.Cause(c.err).Error() == (x509.CertificateInvalidError{Reason: x509.Expired}).Error() {
```
Comment (cpuguy83) [inline daemon/cluster/cluster.go]: Why not convert to `CertificateInvalidError` and check that `cErr.Reason == x509.Expired`?
Seems cleaner than the string comparison.

Comment (tonistiigi) [inline daemon/cluster/cluster.go]: PTAL

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by diogomonica** [comment]
LGTM

**Pull request merged**